src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+package page
+
+import (
+	"github.com/speedianet/os/src/domain/dto"
+	"github.com/speedianet/os/src/domain/valueObject"
+	componentForm "github.com/speedianet/os/src/presentation/ui/component/form"
+	componentMappings "github.com/speedianet/os/src/presentation/ui/component/mappings"
+	componentStructural "github.com/speedianet/os/src/presentation/ui/component/structural"
+)
+
+script MappingsIndexLocalState() {
+    document.addEventListener('alpine:init', () => {
+		Alpine.data('mappings', () => ({
+            // Primary states
+            virtualHost: {},
+            mapping: {},
+            resetPrimaryStates() {
+				this.virtualHost = {
+                    'hostname': '',
+                    'type': 'top-level',
+                    'rootDirectory': '',
+                    'parentHostname': null
+                }
+                this.mapping = {
+                    'id': 0,
+                    'path': '',
+                    'matchPattern': 'begins-with',
+                    'targetType': 'url',
+                    'targetValue': '',
+                    'targetHttpResponseCode': 200
+                }
+            },
+			init() {
+				this.resetPrimaryStates();
+			},
+
+            // Auxiliary states
+            isAdvancedSettingsClosed: true,
+            isCreateMappingFromVhost: false,
+            get vhostWithTrailingSlash() {
+                return this.virtualHost.hostname + "/"
+            },
```

Comment:
Hmm, I fail to see when you would use that, but let's see.

Reply:
We discussed this on the call as well. Since it's a getter that customizes the return of `this.virtualHost.hostname`, it should be in the "primary states" rather than the "auxiliary states." Additionally, the function name was missing "hostname." I have adjusted both its position and name.